app/common/lib/ledgerUtil.js

```diff
@@ -183,16 +186,136 @@ const stickyP = (state, publisherKey) => {
   return (result === undefined || result)
 }
 
-module.exports = {
-  shouldTrackView,
-  batToCurrencyString,
-  formattedTimeFromNow,
-  formattedDateFromTimestamp,
-  walletStatus,
-  blockedP,
-  contributeP,
-  visibleP,
-  eligibleP,
-  stickyP,
-  formatCurrentBalance
+const getMediaId = (data, type) => {
+  let id = null
+
+  if (type == null || data == null) {
+    return id
+  }
+
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE:
+      {
+        id = data.docid
+        break
+      }
+  }
+
+  return id
+}
+
+const getMediaKey = (id, type) => {
+  if (id == null || type == null) {
+    return null
+  }
+
+  return `${type.toLowerCase()}_${id}`
+}
+
+const getMediaData = (xhr, type) => {
+  let result = null
+
+  if (xhr == null || type == null) {
+    return result
+  }
+
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE:
+      {
+        const parsedUrl = urlParse(xhr)
+        let query = null
+
+        if (parsedUrl && parsedUrl.query) {
+          query = queryString.parse(parsedUrl.query)
+        }
+        result = query
+        break
+      }
+  }
+
+  return result
+}
+
+const getMediaDuration = (data, type) => {
+  let duration = 0
+  switch (type) {
+    case ledgerMediaProviders.YOUTUBE: {
+      duration = getYoutubeDuration(data)
+      break
+    }
+  }
+
+  return duration
+}
+
+const getYoutubeDuration = (data) => {
+  let time = 0
+
+  if (data == null || data.st == null || data.et == null) {
+    return time
+  }
+
+  const startTime = data.st.split(',')
+  const endTime = data.et.split(',')
+
+  if (startTime.length !== endTime.length) {
+    return time
+  }
+
+  for (let i = 0; i < startTime.length; i++) {
+    time += parseFloat(endTime[i]) - parseFloat(startTime[i])
+  }
+
+  // we get seconds back, so we need to convert it into ms
+  time = time * 1000
+
+  return parseInt(time)
+}
+
+const isMediaProvider = (url) => {
```

Comment:
Would this be better named as `getMediaProvider`?

Reply:
fixed